components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListener.java

```diff
@@ -126,13 +126,16 @@ public boolean doPostCreateApplication(ServiceProvider serviceProvider, String t
         return true;
     }
 
-    public boolean doPostUpdateApplication(ServiceProvider serviceProvider, String tenantDomain, String userName)
+    public boolean doPostUpdateApplication(ServiceProvider serviceProvider, String tenantDomain,
```

Comment:
Since this is a public method I think it's better to go with an overloading approach.

Reply:
+1

also can't we detect whether isClaimConfigChanged within this listener?

Reply:
Also overloading a method just to detect whether claim config has changed doesn't seem right to me.

Also if you think about it whether isClaimConfigChanged is useful only for one particular listener or few not all listeners. So i feel it would be good to detect whether whether isClaimConfigChanged within the listener implementation logic itself